test/xds/xds_client_federation_test.go

```diff
@@ -140,6 +140,75 @@ func (s) TestClientSideFederation(t *testing.T) {
 	}
 }
 
+// TestClientSideFederationWithTDOM tests that federation is supported with new
+// xdstp style names with TD Open Mesh (TDOM) Authority. TD Open Mesh Authority
+// currently only support new style for LDS. So, this test only tests the new
+// style for LDS resources and the old style for other resources.
+func (s) TestClientSideFederationWithTDOM(t *testing.T) {
+	// Start a management server as TDOM authority.
+	const tdomAuth = "traffic-director-global.xds.googleapis.com"
+	tdomAuthServer, err := e2e.StartManagementServer(e2e.ManagementServerOptions{})
+	if err != nil {
+		t.Fatalf("Failed to spin up the xDS management server: %v", err)
+	}
+	t.Cleanup(tdomAuthServer.Stop)
+
+	// Create a bootstrap file in a temporary directory.
+	nodeID := uuid.New().String()
+	bootstrapContents, err := bootstrap.Contents(bootstrap.Options{
+		NodeID:    nodeID,
+		ServerURI: tdomAuthServer.Address,
+		ClientDefaultListenerResourceNameTemplate: fmt.Sprintf("xdstp://%s/envoy.config.listener.v3.Listener/%%s", tdomAuth),
+		Authorities: map[string]string{tdomAuth: tdomAuthServer.Address},
+	})
+	if err != nil {
+		t.Fatalf("Failed to create bootstrap file: %v", err)
+	}
+
+	resolverBuilder := internal.NewXDSResolverWithConfigForTesting.(func([]byte) (resolver.Builder, error))
+	resolver, err := resolverBuilder(bootstrapContents)
+	if err != nil {
+		t.Fatalf("Failed to create xDS resolver for testing: %v", err)
+	}
+	server := stubserver.StartTestService(t, nil)
+	defer server.Stop()
+
+	const serviceName = "my-service-client-side-xds"
```

Comment:
This string looks like it would have worked before your changes. Does this test fail without your changes to xds_resolver.go?

Reply:
Fixed. Now the serviceName has 2 special chars -  ` `, and `/`.